lib/src/json_schema/json_schema.dart

```diff
@@ -630,9 +630,19 @@ class JsonSchema {
             }
           }
 
-          // If currentSchema is a ref, resolve ref recursively.
-          // Cycle detection happens at evaluation time for drafts 2019 and 2020.
-          if (currentSchema.schemaVersion < SchemaVersion.draft2019_09 && currentSchema.ref != null) {
+          // If currentSchema is a ref, we might want to resolve it recursively right now.
+          if (currentSchema.ref != null) {
+            // If the next fragment is in the current schema, continue with the current schema instead of resolving the ref.
+            if (i + 1 < fragments.length && currentSchema._schemaMap.containsKey(fragments[i + 1])) {
```

Comment:
Might want to check _schemaBool as well

Reply:
Because `_schemaBool` can only exist by it's self, `.ref` will always be null when the schema is a boolean.

Reply:
Fair point